tests/test_diff.py

```diff
@@ -0,0 +1,63 @@
+# Copyright (c) 2019 The Regents of the University of Michigan
+# All rights reserved.
+# This software is licensed under the BSD 3-Clause License.
+import os
+import unittest
+import signac
+from tempfile import TemporaryDirectory
+
+
+class BaseDiffTest(unittest.TestCase):
+
+    project_class = signac.Project
+
+    def setUp(self):
+        self._tmp_dir = TemporaryDirectory(prefix='signac_')
+        self.addCleanup(self._tmp_dir.cleanup)
+        self._tmp_pr = os.path.join(self._tmp_dir.name, 'pr')
+        self._tmp_wd = os.path.join(self._tmp_dir.name, 'wd')
+        os.mkdir(self._tmp_pr)
+        self.config = signac.common.config.load_config()
+        self.project = self.project_class.init_project(
+            name='diff_test_project',
+            root=self._tmp_pr,
+            workspace=self._tmp_wd)
+
+    def tearDown(self):
+        pass
+
+
+class DiffTest(BaseDiffTest):
+
+    def test_two_jobs(self):
+        job1 = self.project.open_job({'a': 0, 'b': 1})
+        job2 = self.project.open_job({'a': 0})
+        expected = {str(job1.id): {'b': 1}, str(job2.id): {}}
+        result = signac.diff_jobs(job1, job2)
+        self.assertEqual(expected, result, f'{result} is not {expected}')
```

Comment:
Unfortunately we can't use format strings on Python 3.5, which we still support.
```suggestion
        self.assertEqual(expected, result, '{} is not {}'.format(result, expected))
```